homeassistant/helpers/translation.py

```diff
@@ -91,40 +94,43 @@ def build_resources(
         else:
             domain = component.split(".", 1)[0]
 
-        if domain not in resources:
-            resources[domain] = {}
+        domain_resources = resources.setdefault(domain, {})
 
         # Add the translations for this component to the domain resources.
         # Since clients cannot determine which platform an entity belongs to,
         # all translations for a domain will be returned together.
-        resources[domain].update(translation_cache[component])
+
+        if category is None:
+            domain_resources.update(translation_cache[component])
```

Comment:
Does the translation cache contain keys that are either languages or domains?

Reply:
categories are the top level entries in `strings.json`. `config`, `options`, etc. [A list is here](https://github.com/home-assistant/architecture/blob/master/adr/0009-translations-2.0.md#context)

So the dict is 

```
{ "light": { "device_automation": {…} }
```